src/test/java/com/exasol/adapter/RequestDispatcherTest.java

```diff
@@ -0,0 +1,140 @@
+package com.exasol.adapter;
+
+import static org.hamcrest.Matchers.startsWith;
+import static org.junit.Assert.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.exasol.ExaMetadata;
+import com.exasol.adapter.metadata.*;
+import com.exasol.adapter.request.*;
+import com.exasol.adapter.response.*;
+
+@ExtendWith(MockitoExtension.class)
+class RequestDispatcherTest {
+    private static final String DEFAULT_REQUEST_PARTS = "\"schemaMetadataInfo\" : { \"name\" : \"foo\" }, " //
+            + "\"properties\" : { \"SQL_DIALECT\" : \"DUMMY\" }";
+    private final ExaMetadata metadata = null;
+    @Mock
+    private VirtualSchemaAdapter adapterMock;
```

Comment:
Why mock the interface, when there is a DummyAdapter for testing?

Reply:
Because I need to modify behavior in the tests. For that a mock is more convenient that a stub. Also then I don't have to fear side effects that could be hidden in the stub.